### ISSUE TITLE
english undefined

### ISSUE THREAD

**Issue by jspitz** [plk/biblatex#742]
I have the following issue with biblatex 3.11: When compiling a document that loads the languages english and british (in this order), I get the following error:
```
! Undefined control sequence.
\extrasenglish ...@resetpunct \abx@extras@english 
                                                  \abx@strings@english 
l.10 ...ish}{Hello} \foreignlanguage{english}{Hi!}
```
This does not happen with biblatex 3.10. It also does not happen when using "american" instead of "english". This is after an upgrade to TL 2018, so also a babel update might be relevant.

MWE:
```
\documentclass[english,british,ngerman]{article}
\usepackage[utf8]{inputenc}
\usepackage{babel}

\usepackage[style=authoryear]{biblatex}
\addbibresource{biblatex-examples.bib}

\begin{document}
	
Hallo \foreignlanguage{british}{Hello} \foreignlanguage{english}{Hi!}
\cite{aksin}

\printbibliography
\end{document}
```

**Comment by moewew** [comment]
Thank you very much for that report. I'll hopefully be able to have a look tomorrow.
For now I can only note that `babel` warns us about that
```
Package Babel Warning: The package option `english' should not be used
(Babel)                with a more specific one (like `british') on input line 
4.
```
and that the `babel` maintainer mentions in a recent TeX.SX post (https://tex.stackexchange.com/q/428922/35864) that "The behavior of the `english` language is not well defined (date and hyphenation) if there are other English ‘dialects'".

But this could well be a consequence of the `.lbx` changes in 3.11.

**Comment by moewew** [comment]
Nope, I can't blame this one on `babel`, it is entirely my fault. A suggested fix is at https://github.com/plk/biblatex/compare/dev...moewew:lbxfifth. The problem was that the helper macro defined to avoid adding an `.lbx` file to the file list multiple times could also block a file from going thorough the input-once routine even for the first time. This is unwanted here.

For testing the macros can be patched in the preamble:
```latex
\documentclass[english,british,ngerman]{article}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}
\usepackage[style=authoryear]{biblatex}
\addbibresource{biblatex-examples.bib}

\makeatletter
\protected\long\def\blx@lbx@input@handler@simple#1#2#3#4#5#6{%
  \blx@info@noline{Trying to load #2..}%
  \IfFileExists{#1}
    {\blx@info@noline{... file '#1' found}%
     #3\@@input\@filef@und#4#5%
     \ifcsundef{blx@file@lbx@simple@#1}
       {\listxadd\blx@list@req@stat{#1}%
        \@addtofilelist{#1}%
        \global\cslet{blx@file@lbx@simple@#1}\@empty}
       {}}
    {\blx@info@noline{... file '#1' not found}#6}}

\protected\long\def\blx@lbx@input@handler@once#1#2#3#4#5#6{%
  \ifcsundef{blx@file@lbx@once@#1}
    {\blx@info@noline{Trying to load #2..}%
     \IfFileExists{#1}
       {\blx@info@noline{... file '#1' found}%
        #3\@@input\@filef@und#4#5%
        \ifcsundef{blx@file@lbx@simple@#1}
          {\listxadd\blx@list@req@stat{#1}%
           \@addtofilelist{#1}}
          {}}
       {\blx@info@noline{... file '#1' not found}#6}%
     \global\cslet{blx@file@lbx@once@#1}\@empty
     \global\cslet{blx@file@lbx@simple@#1}\@empty}
    {#5}}
\makeatother

\begin{document}
Hallo \foreignlanguage{british}{Hello \fullcite{ctan}} \foreignlanguage{english}{Hi! \fullcite{ctan}}
\cite{aksin}

\printbibliography
\end{document}
```

This should have not adversely affect other documents, but testing of the changes would be highly appreciated.

**Comment by jspitz** [comment]
Your patch works for me. Thanks!

**Comment by moewew** [comment]
Good, thanks for checking that. If you have more complicated documents with more interesting language setup, testing would be appreciated. The automatic test suite did not report problems with this fix, but it is not very polyglot.

**Comment by jspitz** [comment]
I have tested on one rather complicated document (where the problem initially occurred).

**Comment by moewew** [comment]
The fix has been merged in #743 and should be in the next release.

**Comment by moewew** [comment]
`biblatex` 3.12 with a fix for this issue has been released and is available in MikTeX and TeX live 2018 now. The MWE from the first post compiles fine now.
